packages/dbml-core/src/parse/dbml/src/lib/lexer/lexer.ts

```diff
@@ -0,0 +1,567 @@
+          }
+      }
+      this.closeGap();
+    }
+    this.tokens.push(SyntaxToken.create(SyntaxTokenKind.EOF, this.start, 0));
+    this.gatherTrivia();
+
+    return new Result(this.tokens, this.errors);
+  }
+
+  gatherTrivia() {
+    let startOfLine = true;
+    let triviaList: SyntaxToken[] = [];
+    let lastNonTrivia: SyntaxToken | undefined;
+    const newTokenList: SyntaxToken[] = [];
+    // eslint-disable-next-line no-restricted-syntax
+    for (const token of this.tokens) {
+      if (isTriviaToken(token)) {
+        triviaList.push(token);
+        if (token.kind === SyntaxTokenKind.NEWLINE) {
+          if (lastNonTrivia) {
+            lastNonTrivia.trailingTrivia = triviaList;
+            startOfLine = true;
+            lastNonTrivia = undefined;
+            triviaList = [];
+          }
+        }
+        continue;
+      }
+      if (startOfLine) {
+        token.leadingTrivia = triviaList;
+        newTokenList.push(token);
+        triviaList = [];
+        lastNonTrivia = token;
+        startOfLine = false;
+      } else {
+        lastNonTrivia!.trailingTrivia = triviaList;
+        newTokenList.push(token);
+        triviaList = [];
+        lastNonTrivia = token;
+      }
+    }
+    this.tokens = newTokenList;
+  }
+
+  checkTripleQuote() {
+    return this.peek(0) === "'" && this.peek(1) === "'" && this.peek(2) === "'";
+  }
+
+  singleLineStringLiteral() {
+    this.skip();
+    let string = '';
+    while (!this.isAtEnd() && !(this.peek() === '\n') && !(this.peek() === "'")) {
+      if (this.peek() === '\\') {
+        string += this.escapedString();
+      } else {
+        string += this.advance();
+      }
+    }
+
+    if (this.isAtEnd() || this.peek() === '\n') {
+      this.errors.push(
+        new ParsingError(
+          ParsingErrorCode.INVALID,
+          'Unclosed single-line string literal',
+          this.start - 1,
+          this.current,
+        ),
+      );
+    } else {
+      this.tokens.push(
+        SyntaxToken.create(
+          SyntaxTokenKind.STRING_LITERAL,
+          this.start - 1,
+          this.current - this.start + 2,
+          string,
+        ),
+      );
+      this.advance();
+    }
+  }
+
+  multilineStringLiteral() {
+    this.skip(3);
+    let string = '';
+    while (!this.isAtEnd() && !this.checkTripleQuote()) {
+      if (this.peek() === '\\') {
+        string += this.escapedString();
+      } else {
+        string += this.advance();
+      }
+    }
+
+    if (this.isAtEnd()) {
+      this.errors.push(
+        new ParsingError(
+          ParsingErrorCode.INVALID,
+          'Unclosed multiline string literal',
+          this.start - 3,
+          this.current,
+        ),
+      );
+    } else {
+      this.tokens.push(
+        SyntaxToken.create(
+          SyntaxTokenKind.STRING_LITERAL,
+          this.start - 3,
+          this.current - this.start + 6,
+          string,
+        ),
+      );
+      this.skip(3);
+    }
+  }
+
+  functionExpression() {
```

Comment:
This method logic is quite similar to `quotedVariable`, `singleLineStringLiteral`, etc... Should we have a generic function for all of them?

Reply:
I have added a method called extractString